src/analytics/index.ts

```diff
@@ -1,10 +1,151 @@
-import { createClient } from '@segment/analytics-react-native';
-// @ts-ignore
+import {
+  createClient,
+  JsonMap,
+  SegmentClient,
+} from '@segment/analytics-react-native';
 import { REACT_APP_SEGMENT_API_WRITE_KEY } from 'react-native-dotenv';
+import { TrackingEventProperties, TrackingEvents } from './trackingEvents';
+import { UserProperties } from './userProperties';
+import Routes from '@/navigation/routesNames';
+import { EthereumAddress } from '@/entities';
+import { ethers } from 'ethers';
+import { forEach } from 'lodash';
 
+// TODO: need to generate secure key
+export function secureHmac(value: string) {
+  return ethers.utils.computeHmac(
```

Comment:
Looks like the sig is `computeHmac(sha256, SECRET_KEY, value)` https://docs.ethers.io/v5/api/utils/hashing/#utils-computeHmac